src/AWS.Deploy.Orchestration/LocalUserSettings/LocalUserSettingsEngine.cs

```diff
@@ -0,0 +1,208 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using AWS.Deploy.Common;
+using AWS.Deploy.Common.IO;
+using Newtonsoft.Json;
+
+namespace AWS.Deploy.Orchestration.LocalUserSettings
+{
+    public interface ILocalUserSettingsEngine
+    {
+        Task UpdateLastDeployedStack(string stackName, string projectName, string? awsAccountId, string? awsRegion);
+        Task DeleteLastDeployedStack(string stackName, string projectName, string? awsAccountId, string? awsRegion);
+        Task CleanOrphanStacks(List<string> deployedStacks, string projectName, string? awsAccountId, string? awsRegion);
+        Task<LocalUserSettings?> GetLocalUserSettings();
+    }
+
+    public class LocalUserSettingsEngine : ILocalUserSettingsEngine
+    {
+        private readonly IFileManager _fileManager;
+        private readonly IDirectoryManager _directoryManager;
+
+        private const string LOCAL_USER_SETTINGS_FILE_NAME = "local-user-settings.json";
+
+        public LocalUserSettingsEngine(IFileManager fileManager, IDirectoryManager directoryManager)
+        {
+            _fileManager = fileManager;
+            _directoryManager = directoryManager;
+        }
+
+        /// <summary>
+        /// This method updates the local user settings json file by adding the name of the stack that was most recently used.
+        /// If the file does not exists then a new file is generated.
+        /// </summary>
+        public async Task UpdateLastDeployedStack(string stackName, string projectName, string? awsAccountId, string? awsRegion)
+        {
+            try
+            {
+                if (string.IsNullOrEmpty(projectName))
+                    throw new FailedToUpdateLocalUserSettingsFileException("The Project Name is not defined.");
+                if (string.IsNullOrEmpty(awsAccountId) || string.IsNullOrEmpty(awsRegion))
+                    throw new FailedToUpdateLocalUserSettingsFileException("The AWS Account Id or Region is not defined.");
+
+                var localUserSettings = await GetLocalUserSettings();
+                var lastDeployedStack = localUserSettings?.LastDeployedStacks?
+                    .FirstOrDefault(x => x.Exists(awsAccountId, awsRegion, projectName));
+
+                if (localUserSettings != null)
+                {
+                    if (lastDeployedStack != null)
+                    {
+                        if (lastDeployedStack.Stacks == null)
```

Comment:
It is redundant check, `LastDeployedStack` has constructor that initializes Stacks with a non-nullable List<string>

Reply:
They are not redundant. When we are deserializing them, if they don't exist in the file, they will be null. Ive ran into null exceptions multiple times which is why i added the check.